contracts/StoryProtocolGateway.sol

```diff
@@ -1,503 +1,424 @@
+    /// @param module The address of the module to set the permission for.
+    /// @param selector The selector of the function to be permitted for execution.
+    function _setPermissionForModule(
+        address ipId,
+        SignatureData calldata sigData,
+        address module,
+        bytes4 selector
+    ) internal {
         IIPAccount(payable(ipId)).executeWithSig(
             address(ACCESS_CONTROLLER),
             0,
-            abi.encodeWithSignature("setBatchPermissions((address,address,address,bytes4,uint8)[])", permissionList),
-            signer,
-            deadline,
-            signature
+            abi.encodeWithSignature(
+                "setPermission(address,address,address,bytes4,uint8)",
+                address(ipId),
+                address(this),
+                address(module),
+                selector,
+                AccessPermission.ALLOW
+            ),
+            sigData.signer,
+            sigData.deadline,
+            sigData.signature
         );
     }
 
-    /// @dev Registers an existing NFT into the protocol as an IP Asset derivative.
-    /// @param licenseIds The parent IP asset licenses used to derive the new IP asset.
-    /// @param royaltyContext The bytes-encoded context for royalty policy to process.
-    /// @param tokenContract The address of the contract of the NFT being registered.
-    /// @param tokenId The id of the NFT being registered.
-    /// @param ipName The name assigned to the IP on registration.
-    /// @param contentHash The content hash assigned to the IP on registration.
-    /// @param externalURL An external URI to link to the IP.
-    function _registerDerivativeIp(
-        uint256[] memory licenseIds,
-        bytes memory royaltyContext,
-        address tokenContract,
-        uint256 tokenId,
-        string memory ipName,
-        bytes32 contentHash,
-        string calldata externalURL
-    ) internal returns (address ipId) {
-        bytes memory canonicalMetadata = abi.encode(
-            IP.MetadataV1({
-                name: ipName,
-                hash: contentHash,
-                registrationDate: uint64(block.timestamp),
-                registrant: msg.sender,
-                uri: externalURL
-            })
-        );
-
-        ipId = IP_ASSET_REGISTRY.register(
-            licenseIds,
-            royaltyContext,
-            block.chainid,
-            tokenContract,
-            tokenId,
-            address(metadataResolver),
-            true,
-            canonicalMetadata
-        );
+    /// @dev Sets the metadata for the given IP if metadata is non-empty.
+    /// @param metadata The metadata to set.
+    /// @param ipId The ID of the IP.
+    function _setMetadata(IPMetadata calldata metadata, address ipId) internal {
+        if (
+            keccak256(abi.encodePacked(metadata.metadataURI)) != keccak256("") ||
+            metadata.metadataHash.length != 0 ||
+            metadata.nftMetadataHash.length != 0
```

Comment:
The code probably contains a bug because both `metadataHash` and `nftMetadataHash` are of type `bytes32`. It's not appropriate to check the length of bytes32 data.

Reply:
Resolved with 1ad6121